server/src/test/java/org/elasticsearch/indices/state/CloseIndexIT.java

```diff
@@ -338,6 +340,37 @@ public void testCloseIndexWaitForActiveShards() throws Exception {
         assertIndexIsClosed(indexName);
     }
 
+    public void testNoopPeerRecoveriesWhenIndexClosed() throws Exception {
```

Comment:
@henningandersen Good suggestion. However we can't test that scenario for now since closing a follower index with gaps in sequence number will make all its shard unassigned; hence no peer recovery will be performed.

Reply:
@dnhatn Can you implement the test scenario that you've described for regular indices (instead of follower index)? It will then show that a closed replica index that is missing some docs *IS doing* a file-based recovery.

Reply:
I added a test in https://github.com/elastic/elasticsearch/pull/41400/commits/b50d3f24aff861651e1325d087c849f29ffa6b92.